From 0917da26bf0722fcf40de1c5a61412f23982d131 Mon Sep 17 00:00:00 2001
Message-Id: <0917da26bf0722fcf40de1c5a61412f23982d131.1554881054.git.pmatilai@redhat.com>
From: Ernestas Kulik <ekulik@redhat.com>
Date: Mon, 25 Feb 2019 16:34:06 +0100
Subject: [PATCH] find-debuginfo.sh: Handle position-independent executables

Since file 5.33, PIEs are identified by a new MIME type, meaning that,
currently, for such executables, the .gnu_debugdata section is not
added, even if -m is passed.

Signed-off-by: Ernestas Kulik <ekulik@redhat.com>
---
 scripts/find-debuginfo.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/find-debuginfo.sh b/scripts/find-debuginfo.sh
index 967814509..6e3ba2ce0 100755
--- a/scripts/find-debuginfo.sh
+++ b/scripts/find-debuginfo.sh
@@ -412,6 +412,7 @@ do_file()
     case "$(file -bi "$f")" in
       application/x-sharedlib*) skip_mini=false ;;
       application/x-executable*) skip_mini=false ;;
+      application/x-pie-executable*) skip_mini=false ;;
     esac
     $skip_mini || add_minidebug "${debugfn}" "$f"
   fi
-- 
2.20.1

